pyproject.toml

```diff
@@ -1,11 +1,143 @@
 [build-system]
-# Minimum requirements for the build system to execute.
-requires = ["setuptools>=61.0.0", "setuptools_scm>=7.0.1"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.0.0", "hatch-vcs"]
+build-backend = "hatchling.build"
 
-[tool.setuptools_scm]
-write_to = "src/pyhf/_version.py"
-local_scheme = "no-local-version"
+[project]
+name = "pyhf"
+dynamic = ["version"]
+description = "pure-Python HistFactory implementation with tensors and autodiff"
+readme = "README.rst"
+license = { text = "Apache-2.0" }  # SPDX short identifier
+requires-python = ">=3.8"
+authors = [
+    { name = "Lukas Heinrich", email = "lukas.heinrich@cern.ch" },
+    { name = "Matthew Feickert", email = "matthew.feickert@cern.ch" },
+    { name = "Giordon Stark", email = "gstark@cern.ch" },
+]
+maintainers = [ {name = "The Scikit-HEP admins", email = "scikit-hep-admins@googlegroups.com"} ]
+keywords = [
+    "fitting",
+    "jax",
+    "numpy",
+    "physics",
+    "pytorch",
+    "scipy",
+    "tensorflow",
+]
```

Comment:
This is properly comma-separated in the generated METADATA, whereas it does not appear so in `main`

Reply:
I think that's okay though? 

https://github.com/scikit-hep/pyhf/blob/3eef1fff2e6d4ccf474bbe51e2211cf690752b82/setup.cfg#L11

shows up as

```
physics, fitting, numpy, scipy, tensorflow, pytorch, jax
```

on https://pypi.org/project/pyhf/0.7.0/ so I would assume that this would render the same with hatchling(?).

Reply:
Oh yes, I was making the observation that the old package's METADATA seems off-spec, but clearly nothing's broken due to it ;)